src/renderer/src/hooks/useMapeoDevice.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react'
+import { mapeoDeviceStore, Role, useMapeoDeviceStoreAction } from './stores/mapeoDeviceStore'
+
+export const useMapeoDevice = (deviceId: string) => {
```

Comment:
another suggestion that came to mind: create separate hooks for actions and data. For example, create a hook for just the device data and one for actions that update state based on that device e.g.

```ts
const useMapeoDevice = (deviceId: string) => {
  return mapeoDeviceStore(state => state.devices[deviceId])
}

// No need to pass an id argument here because the actions should already accept an an id argument on their own
const useMapeoDeviceActions = () => {
  // Can pluck out the specific actions if needed
  return mapeoDeviceStore(state => state.actions)
}
```

In addition to avoiding unnecessary memoization in the hooks, this provides flexibility as the composability of the hooks allows them to be read-only, write-only, or read-and-write:

```tsx
function ReadOnlyComponent() {
  const device = useMapeoDevice(someId)
}

function WriteOnlyComponent() {
  // Can destructure based on relevant actions
  const deviceActions = useMapeoDeviceActions()
}

function ReadWriteComponent() {
  const device = useMapeoDevice(someId)
  // Can destructure based on relevant actions
  const deviceActions = useMapeoDeviceActions()
}
```

Reply:
Referring to [this comment](https://github.com/digidem/mapeo-desktop-shell/pull/40#discussion_r1171593803). I do think the memoization is necessary. There are time where we are updating 1 member, but the list of members is mounted (everything is modal), so it would cause alot of unnecssary re-renders. the list can be up to 15 members long, so everytime we update 1 member, it could cause an extra 14 re-renders, when those 14 components have not changed at all

Reply:
in that case, you can limit the memoization to the `useMapeoDevice` suggestion I wrote:

```
const useMapeoDevice = (deviceId: string) => {
  const device = mapeoDeviceStore(state => state.devices[deviceId])
  
  return useMemo(() => device, [device])
}

  return useMemo(() => mapeoDeviceStore(state => state.devices[deviceId]), [deviceId])
}

Reply:
im not 100% sure this will work due object equality. but im going to test it right now.

Reply:
alternatively, I guess that's what the second arg for the selector function is for (which you already have something for) ?